frontend/mobilityai_react/App/Screens/PatientData.js

```diff
@@ -284,25 +342,77 @@ export default class PatientData extends Component {
 
                     {/* Displaying the circle buttons for each activity */}
                     <View style={styles.flexDir}>
-                        <TouchableHighlight onPress={this._onPressButton.bind(this,activityTypes.standing, this.state.movementPercentages.standing.bar)} underlayColor="white">
-                            <Circle activity='Standing' activityIcon='male' iconLib='font-awesome' color={arrayColours.standing} />
-                        </TouchableHighlight>
-
-                        <TouchableHighlight onPress={this._onPressButton.bind(this,activityTypes.sitting, this.state.movementPercentages.sitting.bar)} underlayColor="white">
-                            <Circle activity='Sitting' activityIcon='airline-seat-recline-normal' iconLib='MaterialIcons' color={arrayColours.sitting} />
-                        </TouchableHighlight>
-
-                        <TouchableHighlight onPress={this._onPressButton.bind(this,activityTypes.lyingDown, this.state.movementPercentages.lyingDown.bar)} underlayColor="white">
-                            <Circle activity='Lying Down' activityIcon='bed' iconLib='font-awesome' color={arrayColours.lyingDown} />
-                        </TouchableHighlight>
-
-                        <TouchableHighlight onPress={this._onPressButton.bind(this,activityTypes.walking, this.state.movementPercentages.walking.bar)} underlayColor="white">
-                            <Circle activity='Walking' activityIcon='directions-walk' iconLib='MaterialIcons' color={arrayColours.walking} />
-                        </TouchableHighlight>
-
-                        <TouchableHighlight onPress={this._onPressButton.bind(this,activityTypes.unknown, this.state.movementPercentages.unknown.bar)} underlayColor="white">
+                        {this.state.singleBarView 
+                            ? <TouchableHighlight onPress={this._onPressButton.bind(this,activityTypes.standing, this.state.movementPercentages.standing.bar)} underlayColor="white">
+                                <Circle activity='Standing' activityIcon='male' iconLib='font-awesome' color={arrayColours.standing} />
+                                </TouchableHighlight>
+                            : null
+                        }
```

Comment:
We should factor these activity circles into 2 different components so we don't have a tonne of if nulls.

Reply:
Make an issue to fix this